keyboards/idobo/keymaps/egstad/keymap.c

```diff
@@ -0,0 +1,267 @@
+    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   _______, _______, _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+    _______, _______, _______, _______, _______, _______, RGB_HUD, RGB_HUI, _______, _______, _______, _______, _______, _______, _______, \
+    KC_ASTG, _______, _______, _______, _______, _______, RGB_SAD, RGB_SAI, _______, _______, KC_LEFT, KC_UP,   KC_DOWN, KC_RGHT, _______, \
+    RGB_TOG, _______, _______, _______, _______, _______, RGB_VAD, RGB_VAI, _______, _______, _______, _______, _______, _______, _______, \
+    RESET,   _______, _______, _______, _______, _______, RGB_RMOD,RGB_MOD, _______, _______, KC_MPLY, KC_MRWD, KC_VOLU, KC_VOLD, KC_MFFD  \
+ ),
+};
+
+
+
+
+
+
+/* ==========================================================================
+    LEDs
+   ========================================================================== */
+
+// RGB MODES
+// 1 = Static
+// 2-5 = Breathing
+// 6-8 = Rainbow
+// 9-14 = Swirl
+// 15-20 = Snake
+// 21-24 = Nightrider
+// 25 = Christmas
+// 26-30 = Static Gradient
+
+// Default LED colors
+uint8_t h = 170;
+uint8_t s = 255;
+uint8_t v;
+
+// default animation
+uint8_t rgbMode = RGBLIGHT_MODE_STATIC_LIGHT;
+// boot animation
+uint8_t rgbBootMode = RGBLIGHT_MODE_SNAKE;
+// boot timeout vars
+uint8_t bootComplete = 0;
+int bootTimeoutDuration = 2000;
+int bootTimeout;
+
+
+void init_hsv(void) {
+    // fetch what the brightness was last sesion
+	v = rgblight_get_val();
+    rgblight_sethsv(h,s,v);
+}
+
+// fetch current HSV vals
+void get_hsv(void) {
+	h = rgblight_get_hue();
+	s = rgblight_get_sat();
+	v = rgblight_get_val();
+}
+
+// reset HSV vals
+void reset_hsv(void) {
+    int currentV = rgblight_get_val();
+	rgblight_sethsv(h,s,currentV);
+}
+
+// deterimes when to stop bootup animation
+void bootupAnimation(void) {
+  bootComplete = (timer_elapsed(bootTimeout) > bootTimeoutDuration) ? 1 : 0;
+
+  if (bootComplete) {
+    rgblight_mode(rgbMode);
+  }
+}
+
+
+
+
+
+
+/* ==========================================================================
+    INITIALIZATION FUNCTION
+   ========================================================================== */
+
+void keyboard_post_init_user(void) {
+  // start a timeout
+  bootTimeout = timer_read();
+  // set rgb color
+  init_hsv();
+  // init rgb
+  rgblight_enable();
+  // animate in snake ledGreende
+  rgblight_mode(rgbBootMode);
+};
+
+
+
+
+
+
+/* ==========================================================================
+    ALWAYS RUNNING
+   ========================================================================== */
+void matrix_scan_user(void) {
+    // keep an eye on these layers
+    uint8_t layer = get_highest_layer(layer_state);
+    // handle boot-up sequence
+    bootupAnimation();
+    // watch the brightness for changes
+    v = rgblight_get_val();
+
+    switch (layer) {
+      case _FN:
+        // set leds to white
+        rgblight_sethsv(h,0,v);
```

Comment:
```suggestion
        rgblight_sethsv_noeeprom(h,0,v);
```

Otherwise this is writing to eeprom each cycle.  That may not be problematic but it may cause issues.

Reply:
Also, this could probably go into `layer_state_set_user` instead of here.

Reply:
This keymap profile doesn't utilize `layer_state_set_user`. I tried leveraging that functionality when first developing this feature, but had little luck getting it to properly compile. Outside of stylistic preference, is there an advantage to using it over the current method?